server/src/main/java/org/elasticsearch/index/mapper/RuntimeField.java

```diff
@@ -92,25 +99,33 @@ public Parser(Function<String, RuntimeField.Builder> builderFunction) {
             this.builderFunction = builderFunction;
         }
 
-        RuntimeField parse(String name, Map<String, Object> node, MappingParserContext parserContext)
+        RuntimeField parse(String name,
+                           Map<String, Object> node,
+                           MappingParserContext parserContext,
+                           String parent,
+                           Function<SearchLookup, CompositeFieldScript.LeafFactory> parentScriptFactory)
             throws MapperParsingException {
 
             RuntimeField.Builder builder = builderFunction.apply(name);
             builder.parse(name, parserContext, node);
-            return builder.createRuntimeField(parserContext);
+            return builder.createRuntimeField(parserContext, parent, parentScriptFactory);
         }
     }
 
     /**
      * Parse runtime fields from the provided map, using the provided parser context.
      * @param node the map that holds the runtime fields configuration
      * @param parserContext the parser context that holds info needed when parsing mappings
+     * @param parent the name of the parent field that the field being parsed is part of, if any
+     * @param parentScriptFactory script factory of the parent field that the field being parsed is part of, if any
      * @param supportsRemoval whether a null value for a runtime field should be properly parsed and
      *                        translated to the removal of such runtime field
      * @return the parsed runtime fields
      */
     static Map<String, RuntimeField> parseRuntimeFields(Map<String, Object> node,
                                                         MappingParserContext parserContext,
+                                                        String parent,
```

Comment:
Since 'parent' and 'parentScriptFactory' only apply to `composite` field parsing, it'd be nice to avoid exposing them in these very general methods. If we created a dedicated parsing method for composite fields, could we avoid updating `RuntimeField#parseRuntimeFields` and `RuntimeField#parse`?